libafl/src/executors/inprocess.rs

```diff
@@ -527,6 +522,70 @@ pub fn inprocess_get_input<'a, I>() -> Option<&'a I> {
     unsafe { (GLOBAL_STATE.current_input_ptr as *const I).as_ref() }
 }
 
+use crate::{
+    corpus::{Corpus, Testcase},
+    events::Event,
+    state::HasMetadata,
+};
+
+#[inline]
+#[allow(clippy::too_many_arguments)]
+/// Save sate if it is interesting
+pub fn save_state_for_restart<E, EM, OF, Z>(
+    executor: &mut E,
+    state: &mut E::State,
+    input: &<E::State as UsesInput>::Input,
+    fuzzer: &mut Z,
+    event_mgr: &mut EM,
+    exitkind: ExitKind,
+) where
+    E: HasObservers,
+    EM: EventFirer<State = E::State> + EventRestarter<State = E::State>,
+    OF: Feedback<E::State>,
+    E::State: HasSolutions + HasClientPerfMonitor,
+    Z: HasObjective<Objective = OF, State = E::State>,
+{
+    let observers = executor.observers_mut();
+
+    observers
```

Comment:
The name `save_state_for_restart` doesn't really show that observers would be called, imho we should make this more clear, `run_observers_and_save_state` or something? Or an extra method for running the observers

Reply:
ok